internal/backend/db/dbgorm/projects.go

```diff
@@ -49,9 +49,7 @@ func (gdb *gormdb) deleteProject(ctx context.Context, projectID sdktypes.UUID) e
 
 // delete project, its envs, deployments, sessions and build
 func (gdb *gormdb) deleteProjectAndDependents(ctx context.Context, projectID sdktypes.UUID) error {
-	// NOTE: should be transactional
-
-	db := gdb.db.WithContext(ctx)
+	// NOTE: should be transactional and with context applied
```

Comment:
if it has to be in a transaction, why not start a transaction here ?

Reply:
it's called from transactional context everywhere anyway.
Since we are not that good at maintaining recursive transaction in our gormdb, in all methods where transactional context expected there is a note clarifying this

Reply:
because we are introducing one more transactional layer in our gormdb (to lock mutex) on top of gorm.DB.
Anyhow - it's an internal method, called from several transactions, previously it was cut-n-pasted logic, now extracted to a method. I think it's unnecessarily to introduce more transactions and locking where unneeded.